compliance/test_utils.py

```diff
@@ -33,14 +33,14 @@ def mock_cybersource_wsdl(mocked_responses, settings, service_version=SERVICE_VE
     Mocks the responses to achieve a functional WSDL
     """
     # in order for zeep to load the wsdl, it will load the wsdl and the accompanying xsd definitions
-    with open(f"{DATA_DIR}/CyberSourceTransaction_{service_version}.wsdl", "r") as wsdl:  # noqa: PTH123, UP015
+    with open(f"{DATA_DIR}/CyberSourceTransaction_{service_version}.wsdl") as wsdl:  # noqa: PTH123
```

Comment:
It does not complain but the `open()` function in Python defaults to read mode ("r"), which `pyupgrade` makes explicitly specifying "r" to redundant-open-mode/unnecessary and removes it. So we can go any of the following options:

1. Keep the Explicit "r"
    - We can keep the explicit "r" for readability if that's preferred. Although `pyupgrade` removed it because it's not necessary, adding it back can make the intent clearer to all code readers, especially those who may not be aware of the default behavior.
2. Adopt Python's Default Behavior 
    - We can follow Python's default behavior to avoid redundant open mode. It resolves [redundant-open-modes (UP015 - derived from the pyupgrade)](https://docs.astral.sh/ruff/rules/redundant-open-modes/#redundant-open-modes-up015) which recommends removing unnecessary open modes to reduce confusion.

Note: The first option will be applied consistently to all instances where this change is present and/or explicitly mentioned.